frontend/src/index.tsx

```diff
@@ -38,18 +39,19 @@ const link = createHttpLink({
 
 const authLink = setContext(async (_, { headers }) => {
   // get the authentication token from local storage if it exists
-  let token: string = getLocalStorageObjProperty(
-    AUTHENTICATED_USER_KEY,
-    "accessToken",
-  );
+  let token: string | null = getLocalStorageObjProperty<
+    NonNullable<AuthenticatedUser>,
+    string
+  >(AUTHENTICATED_USER_KEY, "accessToken");
 
   if (token) {
-    const decodedToken: any = jwt.decode(token);
+    const decodedToken = jwt.decode(token) as DecodedJWT;
 
     // refresh if decodedToken has expired
     if (
       decodedToken &&
-      decodedToken.exp > Math.round(new Date().getTime() / 1000)
+      (typeof decodedToken === "string" ||
+        decodedToken.exp > Math.round(new Date().getTime() / 1000))
```

Comment:
```suggestion
        decodedToken.exp <= Math.round(new Date().getTime() / 1000))
```

Feel free to just commit this